web/src/utils/factory-management/syncState.ts

```diff
@@ -37,17 +37,18 @@ export const calculateSyncState = (factory: Factory) => {
 }
 
 export const checkProductSyncState = (factory: Factory) => {
-  // Scan the products and determine if they are in sync with the syncState.
-  // If there are no products, mark the factory out of sync, if already in sync.
-  if (!factory.products.length) {
-    if (factory.inSync) {
-      factory.inSync = false
-    }
+  // If factory has no products or power producers, mark as out of sync.
+  if (!factory.products.length && !factory.powerProducers.length) {
+    factory.inSync = false
```

Comment:
It should probably return here early. Actually, rather than doing `checkProductSyncState` OR `checkPowerProducerSyncState` if both products and powerProducers are missing, there is literally no point in checking the state is there?